docs/modules/ROOT/pages/secretclass.adoc

```diff
@@ -0,0 +1,117 @@
+= `SecretClass`
+
+A `SecretClass` is a cluster-global Kubernetes resource that defines a category of secrets that the Secret Operator knows how to provision.
+
+This is intended to provide an abstraction between how the secret is used ("I need a certificate for my cluster's TLS PKI") and how
+it is provisioned (automatically and generated by the operator's internal CA, provisioned by the cluster administrator, or provisioned by an
+external service such as Hashicorp Vault).
+
+A `SecretClass` looks like this:
+
+[source,yaml]
+----
+apiVersion: secrets.stackable.tech/v1alpha1
+kind: SecretClass
+metadata:
+  name: tls
+spec:
+  backend: # <1>
+    autoTls: # <2>
+      ca:
+        secret:
+          name: secret-provisioner-tls-ca
+          namespace: default
+    # or... <1>
+    k8sSearch: # <3>
+      searchNamespace:
+        pod: {}
+        # or...
+        name: my-namespace
+      secretLabels:
+        type: custom-secret
+----
+<1> Backends are mutually exclusive, only one may be used by each  SecretClass`
+<2> Configures and selects the xref:#backend-autotls[] backend
+<3> Configures and selects the xref:#backend-k8ssearch[] backend
+
+[#backend]
+== Backend
+
+Each `SecretClass` is a associated with a single backend, which dictates the mechanism for issuing that kind of secret.
+
+[#backend-autotls]
+=== `autoTls`
+
+*Format*: xref:#format-tls[]
+
+Issues a TLS certificate signed by the Secret-Operator.
+
+xref:scope.adoc[Scopes] are used to populate the claims (such as `subjectAlternateName`) of the provisioned certificates.
+
+==== Reference
+
+[source,yaml]
+----
+spec:
+  backend:
+    autoTls:
+      ca:
+        secret:
+          name: secret-provisioner-tls-ca
+          namespace: default
+----
+
+`autoTls`:: Declares that the `autoTls` backend is used.
+`autoTls.ca`:: Configures the certificate authority used to issue `Pod` certificates.
+`autoTls.ca.secret`:: Reference (`name` and `namespace`) to a K8s `Secret` object where the CA certificate and key is stored as `ca.crt`
```

Comment:
Do I understand this correctly: It'll create self-signed certificates if the referenced CA doesn't exist?
This seems like a "dangerous" default and super annoying to debug if someone just has a simple typo in there and expects to reference an existing one.

This comment is not about the docs, I know. And if it documents current behavior we should document it, but if my assumption is true then we might want a follow-up ticket.

Reply:
If the CA doesn't exist then it will create a self-signed CA the first time a certificate is issued, and save that into the k8s `Secret` object that you have specified.

The certificate issued to the pod will never be self-signed.

Reply:
Understood, yes. Sorry, I meant CA not certificate.

This is something worth changing IMO.

Reply:
Feels like there is some misalignment here around what the expected use cases would be. Happy to have a discussion here when you're available.